src/server/components/annual-review/controller.ts

```diff
@@ -134,6 +133,7 @@ async function redirectToFormRunner(req: Request, res: Response, next: NextFunct
 export function declarationGetController(req: Request, res: Response, next: NextFunction): void {
   const { listItemRef } = req.params;
   const errorMsg = req.flash("declarationError")[0];
+  const hasUpdates = req.flash("hasUpdates")[0];
```

Comment:
as long as req.flash("prop") isn't accessed, it will stay accessible. In templates you can access it via `messages.hasUpdates`. you don't have to pass it into the render context. 

Although if you are trying to persist answers between pages, use req.session instead. req.flash is volatile by design -- if you refresh the page, req.flash("hasUpdates") / messages.hasUpdates will be empty. It also means you don't have to rely on hidden fields.

Reply:
changes made to use session attribute instead of hidden form field